### PR TITLE
Automated cherry pick of #14607 upstream release 1.1

### PR DESCRIPTION
**Pull request by mikedanese**
Automated cherry pick of #14607 upstream release 1.1

cc @erictune @janetkuo

### CODE REVIEW

File: pkg/kubectl/resource_printer.go [hunk 1]
```diff
@@ -724,15 +724,22 @@ func printReplicationControllerList(list *api.ReplicationControllerList, w io.Wr
...
+	if withNamespace {
+		if _, err := fmt.Fprintf(w, "%s\t", namespace); err != nil {
+			return err
+		}
 	}
 
 	selector, _ := extensions.PodSelectorAsSelector(job.Spec.Selector)
-	_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%d",
-		job.Name,
+	_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%d\n",
```
Comment (janetkuo) [inline pkg/kubectl/resource_printer.go]: We should remove the trailing \n

Comment (janetkuo) [inline pkg/kubectl/resource_printer.go]: See #16332

**Comment by erictune** [comment]
LGTM

**Comment by a-robinson** [comment]
Merge bot currently off due to a lone flake, merging manually.

**Pull request merged**